### PR TITLE
Add development section to our docs

### PR DESCRIPTION
**Pull request by FabioRosado**
This is the first pass at adding a `development` section for our docs, I've moved `setting up your env` and `developing` from contributing.md into two specific pages. The section in contributing was replaced by a shorter one that also points to these pages in the docs.

The development section is not included in the navbar, should it? I thought that this might be too specific and users might not care if they aren't creating PRs :thinking: 

Please let me know if you would like me to change anything :smile: 

Closes: #907
Fixes: #448

### CODE REVIEW

File: docs/development/developing.md
Comment (antocuni) [inline docs/development/developing.md:50]: I'm not really sure that it's a good idea to try to teach `git` in a few paragraphs:
- for people who knows how to use it, it's useless
- for people who don't, it's probably too short and I'm sure they will run into all sort of corner cases which are not explained here.

Maybe we can put a link to some external tutorial/documentation instead?

(This is a genuine question, I don't really know what's the best way to proceed)

**Review by antocuni — APPROVED** [review]
LGTM

File: docs/development/developing.md
Comment (FabioRosado) [reply docs/development/developing.md:50]: Yeah that's a good point! when I started using Git, rebase was confusing to use that's why I added the section but agree with you, if someone doesn't know git that well, it's likely that they will hit weird cases.

I've added a link to the section on rebase from the github docs 😄

**Comment by antocuni** [comment]
Perfect, thank you!
Our "merge etiquette" prescribes that PR authors do the actual merge after getting approvals, so feel free to merge!
(I assume you have the relevant permissions)

**Comment by FabioRosado** [comment]
Sounds good, I'll hit merge then!

**Pull request merged**
